src/NUnitFramework/framework/Internal/GenericMethodHelper.cs

```diff
@@ -109,7 +109,9 @@ private void TryApplyArgType(Type parmType, Type argType)
             }
             else if (parmType.GetTypeInfo().ContainsGenericParameters)
             {
-                var genericArgTypes = parmType.GetGenericArguments();
+                var genericArgTypes = parmType.IsArray
+                    ? new[] { parmType.GetElementType() }
+                    : parmType.GetGenericArguments();
```

Comment:
This is good. Alternatively, we could fix the existing misnomer (arrays aren't technically generic) by using `parmType.GetElementType()` instead of `genericArgTypes[0]` below inside `if (argType.HasElementType)` and then moving the `var genericArgTypes` inside the `if argType.GetTypeInfo().IsGenericType ...`

Reply:
If I change it to
```csharp
                if (argType.HasElementType)
                {
                    ApplyArgType(parmType.GetElementType(), argType.GetElementType());
                }
                else if (argType.GetTypeInfo().IsGenericType && IsAssignableToGenericType(argType, parmType))
                {
                    var genericArgTypes = parmType.GetGenericArguments();
                    Type[] argTypes = argType.GetGenericArguments();

                    if (argTypes.Length == genericArgTypes.Length)
                        for (int i = 0; i < genericArgTypes.Length; i++)
                            TryApplyArgType(genericArgTypes[i], argTypes[i]);
                }
```
then two of existing tests fails
```
1) Error : NUnit.Framework.Internal.GenericMethodHelperTests.GetTypeArgumentsForMethodTests("MethodWithGenericEnumerableOfType",[Int32[]],[System.Int32])
System.NullReferenceException : Object reference not set to an instance of an object.
   at NUnit.Framework.Internal.GenericMethodHelper.ApplyArgType(Type parmType, Type argType) in C:\src\NUnit\nunit.nunit\src\NUnitFramework\framework\Internal\GenericMethodHelper.cs:line 131
   at NUnit.Framework.Internal.GenericMethodHelper.TryApplyArgType(Type parmType, Type argType) in C:\src\NUnit\nunit.nunit\src\NUnitFramework\framework\Internal\GenericMethodHelper.cs:line 114
   at NUnit.Framework.Internal.GenericMethodHelper.TryGetTypeArguments(Object[] argList, Type[]& typeArguments) in C:\src\NUnit\nunit.nunit\src\NUnitFramework\framework\Internal\GenericMethodHelper.cs:line 87
   at NUnit.Framework.Internal.GenericMethodHelperTests.GetTypeArgumentsForMethodTests(String methodName, Object[] args, Type[] typeArgs) in C:\src\NUnit\nunit.nunit\src\NUnitFramework\tests\Internal\GenericMethodHelperTests.cs:line 116

2) Error : NUnit.Framework.Internal.GenericMethodHelperTests.GetTypeArgumentsForMethodTests("MethodWithGenericEnumerableOfTypeAsSecondArg",["X", Int32[]],[System.String, System.Int32])
System.NullReferenceException : Object reference not set to an instance of an object.
   at NUnit.Framework.Internal.GenericMethodHelper.ApplyArgType(Type parmType, Type argType) in C:\src\NUnit\nunit.nunit\src\NUnitFramework\framework\Internal\GenericMethodHelper.cs:line 131
   at NUnit.Framework.Internal.GenericMethodHelper.TryApplyArgType(Type parmType, Type argType) in C:\src\NUnit\nunit.nunit\src\NUnitFramework\framework\Internal\GenericMethodHelper.cs:line 114
   at NUnit.Framework.Internal.GenericMethodHelper.TryGetTypeArguments(Object[] argList, Type[]& typeArguments) in C:\src\NUnit\nunit.nunit\src\NUnitFramework\framework\Internal\GenericMethodHelper.cs:line 87
   at NUnit.Framework.Internal.GenericMethodHelperTests.GetTypeArgumentsForMethodTests(String methodName, Object[] args, Type[] typeArgs) in C:\src\NUnit\nunit.nunit\src\NUnitFramework\tests\Internal\GenericMethodHelperTests.cs:line 116
```
the problem is that `argType.HasElementType` is `true` (`argType` is `Int32[]`), but `parmType.GetElementType()` returns to `null`, as `parmType` is ``IEnumerable`1``.

Reply:
Oh! I missed that `argType` and `parmType` weren't the same variable. Thanks for the explanation!